Source/WebKit/UIProcess/wpe/WebPasteboardProxyWPE.cpp

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2025 Igalia S.L.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebPasteboardProxy.h"
+
+#if PLATFORM(WPE)
+#include "Connection.h"
+#include <WebCore/Pasteboard.h>
+#include <WebCore/PasteboardCustomData.h>
+#include <WebCore/PasteboardItemInfo.h>
+#include <WebCore/PlatformPasteboard.h>
+#include <WebCore/SelectionData.h>
+
+#if ENABLE(WPE_PLATFORM)
+#include <wpe/wpe-platform.h>
+#endif
+
+namespace WebKit {
+using namespace WebCore;
+
+void WebPasteboardProxy::getTypes(const String& pasteboardName, CompletionHandler<void(Vector<String>&&)>&& completionHandler)
+{
+#if ENABLE(WPE_PLATFORM)
+    if (g_type_class_peek(WPE_TYPE_DISPLAY)) {
+        completionHandler({ });
+        return;
+    }
+#endif
```

Comment:
So I understand the intent here is "do not execute this code if the new WPE Platform API is actually being used.

But two problems. First, this is not obvious. You can make it obvious and self-documenting by adding a simple function:

```
bool isWPEPlatformAPIEnabled()
{
#if ENABLE(WPE_PLATFORM)
    return g_type_class_peek(WPE_TYPE_DISPLAY);
#else
    return false;
#endif
}
```

OK, so that's better, but second problem is: why bail early when the new API is enabled? The pasteboard classes are too complicated and it's hard for me to understand. Is this future work, or is WebPasteboardProxy not expected to be used for some reason...?

Reply:
Yes, this is in preparation for the clipboard support with the new api. We currently crash because we try to use libwpe, so for now the early return fixes the crashes, but it will include the actual implementation in a follow up patch.